File: backend/src/main/java/com/starttostar/backend/service/MessageService.java
package com.starttostar.backend.service;

import com.starttostar.backend.entity.Message;
import com.starttostar.backend.repository.MessageRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class MessageService {

    private final MessageRepository messageRepository;

    public MessageService(MessageRepository messageRepository) {
        this.messageRepository = messageRepository;
    }

    // SEND MESSAGE
    public Message sendMessage(Message message) {
        if (message.getCreatedAt() == null) {
            message.setCreatedAt(LocalDateTime.now());
        }
        if (message.getIsRead() == null) {
            message.setIsRead(false);
        }
        return messageRepository.save(message);
    }

    // GET USER MESSAGES
    public List<Message> getUserMessages(String email) {
        return messageRepository.findBySenderEmailOrReceiverEmailOrderByCreatedAtAsc(
                email,
                email
        );
    }

    // GET ALL MESSAGES
    public List<Message> getAllMessages() {
        return messageRepository.findAll();
    }

    // DELETE MESSAGE
    public boolean deleteMessage(Long id) {
        if (messageRepository.existsById(id)) {
            messageRepository.deleteById(id);
            return true;
        }
        return false;
    }
}

File: client/src/services/MessageService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/messages";

// SEND MESSAGE
export const sendMessage = (message) => {
  return axios.post(API_URL, message);
};

// GET USER MESSAGES
export const getUserMessages = (email) => {
  return axios.get(`${API_URL}/${email}`);
};

// GET ALL MESSAGES
export const getAllMessages = () => {
  return axios.get(API_URL);
};

// DELETE MESSAGE
export const deleteMessage = (id) => {
  return axios.delete(`${API_URL}/${id}`);
};